src/js/containers/search/filters/naics/NAICSContainer.jsx

```diff
@@ -191,6 +197,13 @@ export class NAICSContainer extends React.Component {
             });
     }
 
+    componentDidUpdate(prevProps) {
+        if (this.props.checked.length === 0 && prevProps.checked.length !== 0 && this.state.stagedNaicsFilters.length !== 0) {
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({ stagedNaicsFilters: [] });
```

Comment:
I tried to add a test for this but simulating this scenario is hard to do... Also, it's gonna be refactored. Would like to prioritize this over creating the checkbox tree component for the library.